From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Sat, 22 Mar 2014 23:52:54 -0400
Subject: [PATCH] ZombieReinforcementEvent

Control when a zombie reinforcement event should occur (modify chance)
also makes it run on all difficulty levels

diff --git a/src/main/java/net/minecraft/world/entity/monster/Zombie.java b/src/main/java/net/minecraft/world/entity/monster/Zombie.java
index e72e9b748b3f3e34baddf01366c703efba50c67c..3b91de97bf64249a6e28138848367a61ba20c65e 100644
--- a/src/main/java/net/minecraft/world/entity/monster/Zombie.java
+++ b/src/main/java/net/minecraft/world/entity/monster/Zombie.java
@@ -321,7 +321,17 @@ public class Zombie extends Monster {
                 entityliving = (LivingEntity) source.getEntity();
             }
 
-            if (entityliving != null && this.level.getDifficulty() == Difficulty.HARD && (double) this.random.nextFloat() < this.getAttributeValue(Attributes.SPAWN_REINFORCEMENTS_CHANCE) && this.level.getGameRules().getBoolean(GameRules.RULE_DOMOBSPAWNING)) {
+            // EMC start
+            boolean spawnReinforcements = false;
+            if (entityliving != null && this.level.getGameRules().getBoolean(GameRules.RULE_DOMOBSPAWNING)) {
+                final double chance = this.getAttributeValue(Attributes.SPAWN_REINFORCEMENTS_CHANCE);
+                com.empireminecraft.customevents.ZombieReinforcementEvent event = new com.empireminecraft.customevents.ZombieReinforcementEvent( this.getBukkitEntity(), (org.bukkit.entity.LivingEntity) entityliving.getBukkitEntity(), chance);
+                if (event.callEvent()) {
+                    spawnReinforcements = this.random.nextFloat() < event.getChance();
+                }
+            }
+            if (spawnReinforcements) {
+                // EMC end
                 int i = Mth.floor(this.getX());
                 int j = Mth.floor(this.getY());
                 int k = Mth.floor(this.getZ());
